base/parse.jl

```diff
@@ -217,6 +217,82 @@ function parse(::Type{T}, s::AbstractString; base::Union{Nothing,Integer} = noth
 end
 
 ## string to float functions ##
+function parsefloat_preamble(s::AbstractString, base::Int, i::Int=1)
+    isempty(s) && throw(ArgumentError("Empty string!"))
+    c = ' '
+    sign = 1
+    while isspace(c)
+        c, i = next(s,i)
+    end
+    if c in ('-', '+')
+        sign = (c == '-') ? -1 : 1
+        c, i = next(s, i)
+    end
+    while isspace(c)
+        c, i = next(s,i)
+    end
+    startpos = i-1
+    return sign, startpos
+end
+
+function parse{T<:AbstractFloat}(::Type{T}, s::AbstractString, base::Integer)
+    sign, i = parsefloat_preamble(s, base)
+
+    if !done(s, i+2)
+        nan = [false, false, false]
+        inf = [false, false, false]
+        for j in 1:3
+            c, _ = next(s, i+j-1)
+            nan[j] = lowercase(c) == "nan"[j]
+            inf[j] = lowercase(c) == "inf"[j]
+        end
+        all(nan) && return T(NaN)
+        all(inf) && return sign*T(Inf)
+    end
+
+    baseunder15 = base < 15 # only interpret 'e' as scientific notation if unambiguous
+    b = T(base)
+    res = zero(T)
+    while !done(s, i)
+        c, i = next(s, i)
+        isspace(c) && break
+        c == '.' && break
+        if baseunder15 && c == 'e'
+            i -= 1
+            break
+        end
+        tmp = parse(Int, c, base=base)
+
+        res = res * b + tmp
+    end
+
+    n = 0
+    while !done(s, i)
+        n -= 1
+        c, i = next(s, i)
+        isspace(c) && break
+        baseunder15 && c == 'e' && break
+        tmp = parse(Int, c, base=base)
+
+        res += tmp * b^n
+    end
+
+    expsign = 1
+    exponent = 0
+    while baseunder15 && !done(s, i)
+        c, i = next(s, i)
+        isspace(c) && break
+        if c == '-'
+            expsign = -1
+            continue
+        end
+        tmp = parse(Int, c, base=base)
+
+        exponent = exponent * base + tmp
+    end
+
+    return sign*res*b^(expsign*exponent)
```

Comment:
```julia
julia> parse(Float64, "1.2e2 some nonesense", 10)
120.0
```
Should throw an error instead, like here:
```
julia> parse(Float64, "1.2e2 some nonesense")
ERROR: ArgumentError: cannot parse "1.2e2 some nonesense" as Float64
```
Probably by checking that the rest of `s` is whitespace.

Reply:
Also 
```
julia> parse(Float64, "nanana")
ERROR: ArgumentError: cannot parse "nanana" as Float64
Stacktrace:
 [1] tryparse_internal(::Type{Float64}, ::String, ::Int64, ::Int64, ::Bool) at ./parse.jl:410
 [2] parse(::Type{Float64}, ::String) at ./parse.jl:417
 [3] top-level scope

julia> parse(Float64,  "nanana", 10)
NaN
```